kasa/smartdevice.py

```diff
@@ -760,10 +777,15 @@ def internal_state(self) -> Any:
         The returned object contains the raw results from the last update call.
         This should only be used for debugging purposes.
         """
-        return self._last_update
+        return self._last_update or self._discovery_info
 
     def __repr__(self):
         if self._last_update is None:
+            if self._discovery_info:
+                return (
+                    f"<{self._device_type} model {self.model} at {self.host}"
+                    f" ({self.alias})"
+                )
```

Comment:
See my comments below wrt. filling `_info` with discovery data. This will become unnecessary (except maybe adding a note that the device has just been found but not initialized by update yet).

Reply:
Fixed